components/tensorflow-notebook-image/start.sh

```diff
@@ -24,14 +24,11 @@ if [ $(id -u) == 0 ] ; then
     if [ "$NB_UID" != $(id -u $NB_USER) ] ; then
         echo "Set user UID to: $NB_UID"
         usermod -u $NB_UID $NB_USER
-        # Careful: $HOME might resolve to /root depending on how the
-        # container is started. Use the $NB_USER home path explicitly.
-        for d in "$CONDA_DIR" "$JULIA_PKGDIR" "/home/$NB_USER"; do
-            if [[ ! -z "$d" && -d "$d" ]]; then
-                echo "Set ownership to uid $NB_UID: $d"
-                chown -R $NB_UID "$d"
-            fi
-        done
+    fi
+
+    # Ensure pv directory belongs to user
```

Comment:
Why are you no longer changing the ownership of the CONDA_DIR and JULIA_PKG_DIR.

Can you explain what the new code does differently? It looks like the only difference is that instead of doing chown with the UID we use the username now.

Do you know why that makes a difference?

Reply:
Difference is that, instead of changing only CONDA_DIR and JULIA_PKG_DIR (both of which lives in /home/$NB_USER) we chown whole /home/$NB_USER - most importantly /home/$NB_USER/work

Because we change user here in the script (line 48) and not in kubernetes manifest, PV isn't connected with correct user ownership, it assumes root (if you look at our notebook pod manifesto, it explicitly runs container as UID 0). I'm doing chown in startup script instead.

Reply:
Do you know why run the container as UID 0? Should we not be doing that?

The original code was 

```
for d in "$CONDA_DIR" "$JULIA_PKGDIR" "/home/$NB_USER"; do
```

So it chown'd the whole /home/$NB_USER dir.

It looks like the difference is that you are now doing this always as opposed to

```
if [ "$NB_UID" != $(id -u $NB_USER) ]
```
What I'm inferring from this is that NB_UID = id NB_USER therefore in the current startup script we are not executing the chown. 

However, for some reason the permsions on /home/${NB_USER} are not set correctly to allow that user to use that container.

If UID is 0, doesn't that mean we are running as root?

Reply:
Ah you're right, I'll restore CONDA_DIR etc, issue was "if" above.

I think we shouldn't run it as root, maybe ensure perms for these dirs in Dockerfile. I'll open separate issue for that one tho as someone did that on purpose:)

Yeah we run pod as root, but start.sh does su $NB_USER <<command>> so we run jupyter itself as NB_USER. Not a typical way to deal with it tbh.